test/rules/flutter_style_todos_test.dart

```diff
@@ -0,0 +1,45 @@
+// Copyright (c) 2022, the Dart project authors. Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'package:test_reflective_loader/test_reflective_loader.dart';
+
+import '../rule_test_support.dart';
+
+main() {
+  defineReflectiveSuite(() {
+    defineReflectiveTests(FlutterStyleTodosTest);
+  });
+}
+
+@reflectiveTest
+class FlutterStyleTodosTest extends LintRuleTest {
+  @override
+  String get lintRule => 'flutter_style_todos';
+
+  test_main() async {
+    await assertDiagnostics(
+      r'''
+// TODO something
+// Todo something
+// todo something
+// TODO(somebody) something
+// TODO: something
+// Todo(somebody): something
+// todo(somebody): something
+// ToDo(somebody): something
+// TODO(somebody): something
+''',
+      [
```

Comment:
I can't actually tell from looking at this list where the expected failures are.  Could you consider (at least) breaking these up into one test for OK patterns and another for bad ones?